src/SportsHub.Infrastructure/Repositories/TeamRepository.cs

```diff
@@ -74,5 +73,34 @@ public async Task<bool> DoesTeamAlredyExistByIdAsync(Guid id)
 
             return teams.Any(team => team.Id == id);
         }
+
+        public IEnumerable<Team> GetTeamsFilteredByLocation(string location, ICollection<Team> teams)
+        {
+            return teams.Where(teams => teams.Location == location).ToList();
+        }
+
+        public IEnumerable<Team> GetTeamsFilteredBySubcategoryId(Guid subcategoryId, ICollection<Team> teams)
+        {
+            return teams.Where(teams => teams.SubcategoryId == subcategoryId).ToList();
+        }
+
+        public IEnumerable<Team> GetTeamsFilteredBySubcategoryIds(IEnumerable<Subcategory> subcategories, ICollection<Team> teams)
+        {
+            var subcategoryIds = subcategories.Select(subcategory => subcategory.Id);
+            var newTeam = teams;
+            teams = newTeam.Where(team => team.SubcategoryId == subcategoryIds.ToList()[0]).ToList();
```

Comment:
Why did you make this variable?
It's a reference type, so they'll have the same value within, so it's makes no sense having this variable